libs/MeshKernel/src/Mesh2DGenerateGlobal.cpp

```diff
@@ -0,0 +1,198 @@
+#include "MeshKernel/Mesh2DGenerateGlobal.hpp"
+#include <MeshKernel/Operations.hpp>
+#include <MeshKernel/Polygons.hpp>
+#include <cmath>
+
+using namespace meshkernel;
+
+double Mesh2DGenerateGlobal::DeltaLatitude(const double currentLatitude, const double longitudeDiscretization)
+{
+
+    double deltaLatitude = longitudeDiscretization * std::cos(constants::conversion::degToRad * currentLatitude);
+
+    for (UInt i = 0; i < numIterations; ++i)
+    {
+        const double phi = constants::conversion::degToRad * (currentLatitude + 0.5 * deltaLatitude);
+        const double cosPhi = std::cos(phi);
+        const double sinPhi = std::sqrt(1.0 - cosPhi * cosPhi);
+        const double f = deltaLatitude - longitudeDiscretization * cosPhi;
+        const double df = 1.0 + 0.5 * constants::conversion::degToRad * longitudeDiscretization * sinPhi;
+        const double latitudeAdjustment = f / df;
+        deltaLatitude = deltaLatitude - latitudeAdjustment;
+
+        if (latitudeAdjustment < tolerance)
+        {
+            break;
+        }
+    }
+
+    return deltaLatitude;
+}
+
+UInt Mesh2DGenerateGlobal::NodeIndexFromPosition(const Mesh& mesh, const Point& x)
+{
+    constexpr double tolerance = 1.0e-6;
+    const auto nodeIndex = constants::missing::uintValue;
+
+    for (int i = static_cast<int>(mesh.m_nodes.size() - 1); i >= 0; --i)
+    {
+        if (IsEqual(x, mesh.m_nodes[i], tolerance))
+        {
+            return static_cast<UInt>(i);
+        }
+    }
+    return nodeIndex;
+}
+
+void Mesh2DGenerateGlobal::AddFace(Mesh& mesh,
+                                   const std::array<Point, 5>& points,
+                                   const GridExpansionDirection growingDirection,
+                                   const UInt numNodes)
+{
+    std::array<UInt, 5> nodeIndices{};
+
+    for (UInt n = 0; n < numNodes; ++n)
+    {
+        const double expansionMultiplier = (growingDirection == GridExpansionDirection::Northwards) ? 1.0 : -1.0;
```

Comment:
if the enum is adapted this line is not needed and you can multiply directly by `static_cast<double>(growingDirection)`